### PR TITLE
Add io.github.koromelodev.mindmate

### PR DESCRIPTION
**Pull request by KoromeloDev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.koromelodev.mindmate.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+app-id: org.koromelodev.mindmate
```
Comment (hfiguiere) [inline org.koromelodev.mindmate.yml:2]: `koromelodev.org` doesn't exist. So the app-id doesn't meet the requirements.

File: org.koromelodev.mindmate.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+runtime: org.kde.Platform
+runtime-version: '6.4'
+sdk: org.kde.Sdk
+command: MindMate
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline org.koromelodev.mindmate.yml:10]: does it suppor wayland? If yes then it should be `fallback-x11`

File: org.koromelodev.mindmate.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  - --socket=pulseaudio
+  - --filesystem=host
```
Comment (hfiguiere) [inline org.koromelodev.mindmate.yml:12]: this is a very broad filesystem permission

File: org.koromelodev.mindmate.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  - --device=dri
+  - --share=network
+modules:
+  - name: MindMate
+    buildsystem: cmake
```
Comment (hfiguiere) [inline org.koromelodev.mindmate.yml:17]:
```suggestion
    buildsystem: cmake-ninja
```

unless it doesn't work.

**Comment by hfiguiere** [comment]
the app-id in the appstream file is incorrect.

File: io.github.koromelodev.mindmate.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+command: MindMate
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --device=dri
+  - --share=network
+  - --persist=/.var/app/io.github.koromelodev.mindmate/config
```
Comment (hfiguiere) [inline io.github.koromelodev.mindmate.yml:15]: you don't need that.
```suggestion
```

File: io.github.koromelodev.mindmate.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+app-id: io.github.koromelodev.mindmate
+runtime: org.kde.Platform
+runtime-version: '6.4'
+sdk: org.kde.Sdk
+command: MindMate
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline io.github.koromelodev.mindmate.yml:11]:
```suggestion
  - --socket=wayland
  - --socket=fallback-x11
```

**Comment by hfiguiere** [comment]
bot, build io.github.koromelodev.mindmate

**Comment by hfiguiere** [comment]
> the app-id in the appstream file is incorrect.

This is the current build error.

File: org.koromelodev.mindmate.yml (hunk 4 above)
Comment (hfiguiere) [reply org.koromelodev.mindmate.yml]: this wasn't adressed.

**Comment by KoromeloDev** [comment]
bot, build io.github.koromelodev.mindmate

**Comment by KoromeloDev** [comment]
bot, build io.github.koromelodev.mindmate

File: io.github.koromelodev.mindmate.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --device=dri
+  - --share=network
+modules:
+  - name: MindMate
+    buildsystem: cmake
```
Comment (hfiguiere) [inline io.github.koromelodev.mindmate.yml:16]:
```suggestion
    buildsystem: cmake-ninja
```

**Comment by barthalion** [comment]
```
builddir/files/share/appdata/io.github.koromelodev.mindmate.appdata.xml: FAILED:
• tag-missing           : <release> required
• tag-missing           : <url> is not present
```

**Comment by KoromeloDev** [comment]
bot, build io.github.koromelodev.mindmate

**Comment by KoromeloDev** [comment]
bot, build io.github.koromelodev.mindmate

**Comment by hfiguiere** [comment]
the launchable value in the appstream file is incorrect.

**Comment by KoromeloDev** [comment]
bot, build io.github.koromelodev.mindmate

File: io.github.koromelodev.mindmate.yml [hunk 8]
```diff
@@ -20,4 +20,4 @@ modules:
     sources:
       - type: archive
         url: https://github.com/KoromeloDev/MindMate/archive/refs/tags/v1.1.9-beta.tar.gz
-        sha256: e22ed0d84c5cca2425b04d519ade1fc8474a032423afd28672f8fc456ba8393f
+        sha256: e8624ff341fd51e1da10c2e10475a6d1079e0409cccf11f387f8f5726fd56bed
```
Comment (hfiguiere) [inline io.github.koromelodev.mindmate.yml:23]: for the record, retagging is a bad practice.

File: io.github.koromelodev.mindmate.yml (hunk 7 above)
Comment (hfiguiere) [reply io.github.koromelodev.mindmate.yml:16]: you have ignored this.

**Comment by hfiguiere** [comment]
The installed desktop file is:
```
-- Installing: /app/share/applications/MindMate.desktop
```

You can use `rename-desktop-file`.

**Comment by KoromeloDev** [comment]
bot, build io.github.koromelodev.mindmate

**Comment by KoromeloDev** [comment]
bot, build io.github.koromelodev.mindmate

**Comment by hfiguiere** [comment]
the desktop file needs to be renamed using `rename-desktop-file` as I already mentionned in https://github.com/flathub/flathub/pull/4171#issuecomment-1568392088

**Pull request closed without merging**